Library/Homebrew/cask/installer.rb

```diff
@@ -104,14 +105,14 @@ def self.caveats(cask)
     def fetch(quiet: nil, timeout: nil)
       odebug "Cask::Installer#fetch"
 
-      load_cask_from_source_api! if @cask.loaded_from_api? && @cask.caskfile_only?
+      load_cask_from_source_api! if cask_from_source_api?
       verify_has_sha if require_sha? && !force?
       check_requirements
 
       forbidden_tap_check
       forbidden_cask_and_formula_check
 
-      download(quiet:, timeout:)
+      download(quiet:, timeout:) if @download_queue.nil?
```

Comment:
This seems a little weird to live inside `fetch`. Is it worth moving most of this method to `prefetch` and then having `fetch` only be called when using the download queue?